src/token-group/dismiss-button.tsx

```diff
@@ -38,13 +40,12 @@ function DismissButton(
         if (disabled || readOnly || !onDismiss) {
           return;
         }
-
         onDismiss();
       }}
       aria-label={dismissLabel}
       {...(disabled || readOnly ? {} : getAnalyticsMetadataAttribute(analyticsMetadata))}
     >
-      <InternalIcon name="close" />
+      {loading ? <InternalSpinner /> : <InternalIcon name="close" />}
```

Comment:
Why is spinner rendered instead of the close button? I think removing files while uploading is a legitimate use case.